subsys/pm/device_runtime.c

```diff
@@ -145,6 +145,10 @@ int pm_device_runtime_get(const struct device *dev)
 	}
 
 	if (!k_is_pre_kernel()) {
+		if (k_is_in_isr()) {
+			return -EWOULDBLOCK;
+		}
+
 		(void)k_sem_take(&pm->lock, K_FOREVER);
```

Comment:
Suggestion:
```
rc = k_sem_take(&pm->lock, k_is_in_isr() ? K_NO_WAIT : K_FOREVER);
if (rc == -EAGAIN) {
  return -EWOULDBLOCK;
}
```

Reply:
good point, I've updated the code and handled the suspending state as well.